lib/tlscontext.c

```diff
@@ -335,6 +338,52 @@ _set_sni_in_client_mode(TLSSession *self)
   return FALSE;
 }
 
+void
+_write_line_to_keylog_file(const char *file_path, const char *line, FILE *keylog_file, GMutex *mutex)
+{
+  gint ret_val;
+  g_mutex_lock(mutex);
+  if(keylog_file)
+    {
+      ret_val = fprintf(keylog_file, "%s\n", line);
+      if (ret_val != strlen(line))
+        msg_error("Couldn't write to TLS keylogfile", evt_tag_errno("error", ret_val));
+
+      fflush(keylog_file);
+    }
+  else
+    {
+      msg_error("Error opening keylog-file",
+                evt_tag_str(EVT_TAG_FILENAME, file_path),
+                evt_tag_error(EVT_TAG_OSERROR));
```

Comment:
This error message is misleading as there is no file opening here at all.
At this point the file is either opened or not, it would be possible to try open if we were not successful in the past.
I would vote not to try open the file here, and in that case there is no need to log anything or do anything at all.

Also I think the check for `keylog_file` can be done outside of the mutex, as it is only changed once when the tls context created.
```c
void
_write_line_to_keylog_file(onst char *file_path, const char *line, FILE *keylog_file, GMutex *mutex)
{
  if (!keylog_file)
      return;

  gint ret_val;
  g_mutex_lock(mutex);
  ...
}
```